File: backend/src/main/java/com/iiitp/attendance/model/Student.java
package com.iiitp.attendance.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "students")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer studentId;

    @Column(nullable = false)
    private String name;

    @Column(unique = true, nullable = false)
    private String rollNo;

    @Column(unique = true, nullable = false)
    private String qrCode;

    private String department;

    @Column(nullable = false)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Role role;

    @Column(unique = true)
    private String email;

    private String otp;
    private java.time.LocalDateTime otpExpiry;

    private Integer year;
    private Integer semester;
    private String hostel;
}

File: frontend/lib/models/student.dart
class Student {
  final int? studentId;
  final String name;
  final String rollNo;
  final String qrCode;
  final String? department;
  final String? password;
  final String? role;

  Student({
    this.studentId,
    required this.name,
    required this.rollNo,
    required this.qrCode,
    this.department,
    this.password,
    this.role,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      studentId: json['studentId'],
      name: json['name'],
      rollNo: json['rollNo'],
      qrCode: json['qrCode'],
      department: json['department'],
      password: json['password'],
      role: json['role'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'studentId': studentId,
      'name': name,
      'rollNo': rollNo,
      'qrCode': qrCode,
      'department': department,
      'password': password,
      'role': role,
    };
  }
}
